lib/nostrum/voice/opus.ex

```diff
@@ -4,58 +4,41 @@ defmodule Nostrum.Voice.Opus do
   use Bitwise
 
   @doc """
-  Strips the RTP extensions from an RTP payload.
+  Strips the RTP header extension from an RTP payload.
 
   The encrypted portion of the RTP packets that we receive from Discord
-  isn't raw opus packets; it's actually opus prepended by RTP header extensions.
+  isn't raw opus packets; it's actually opus prepended by an RTP header extension.
   Looking at the unencrypted fixed-length RTP header, the extension bit is, in
   fact, set to 1.
 
-  Discord seems to have misinterpreted the RTP header extension RFC and
-  thought that for one-byte header extensions that the two-byte integer
-  length following the constant 0xBEDE pattern is equivalent to the number
-  of header extensions - 1 (where ext length 0 would represent 1 extension),
-  but this is only the case for the 4-bit integer that describes each
-  one-byte header extension.
-
-  discord.js is the only other library that implements receiving audio, and
-  in their stripping of the RTP header extensions they account for an extra
-  "Discord byte" but it looks like they are actually incorrectly incrementing
-  the offset by an extra 1, which causes them to land on an extra 1 byte,
+  RFC 8285 describes the format of RTP header extensions. Discord uses RTP
+  extensions with the one-byte header format that begins with a 0xBEDE pattern
+  followed by a 16 bit extension length. This extension length represents the length
+  of the following extension in 32-bit words.
+
+  discord.js appears to be the only other library that officially implements receiving
+  audio as a part of their voice modules, and in their stripping of the RTP header
+  extension they account for an extra "Discord byte" but it looks like they are actually
+  misinterpreting the 16-bit extension lenth as the number of extension elements, then
+  iterating through each one-byte header and variable-length element but incorrectly
+  incrementing the offset by an extra 1, which causes them to land on an extra 1 byte,
   instead a full extension (1 byte header with 1 byte extension).
```

Comment:
thinking about this - should we maybe report it to their devs?

Reply:
Yeah probably